internal/pkg/hub/manifest.go

```diff
@@ -0,0 +1,95 @@
+package hub
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/http/httputil"
+
+	"github.com/confluentinc/cli/internal/pkg/ccstructs"
+	"github.com/confluentinc/cli/internal/pkg/errors"
+	"github.com/confluentinc/cli/internal/pkg/log"
+)
+
+func (c *Client) GetRemoteManifest(owner, name, version string) (*ccstructs.Manifest, error) {
+	manifestUrl := fmt.Sprintf("%s/api/plugins/%s/%s", c.URL, owner, name)
+	if version != "latest" {
+		manifestUrl = fmt.Sprintf("%s/versions/%s", manifestUrl, version)
+	}
+
+	req, err := http.NewRequest(http.MethodGet, manifestUrl, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	if c.Debug {
+		dump, err := httputil.DumpRequestOut(req, true)
+		if err != nil {
+			return nil, err
+		}
+		log.CliLogger.Tracef("\n%s\n", string(dump))
+	}
+
+	r, err := c.Client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	if c.Debug {
+		dump, err := httputil.DumpResponse(r, true)
+		if err != nil {
+			return nil, err
+		}
+		log.CliLogger.Tracef("\n%s\n", string(dump))
+	}
+
+	defer r.Body.Close()
+	body, err := io.ReadAll(r.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	if r.StatusCode != http.StatusOK {
+		response := make(map[string]interface{})
+		_ = json.Unmarshal(body, &response)
+		if errorMessage, ok := response["message"]; ok {
+			return nil, errors.Errorf("failed to read manifest file from Confluent Hub: %s", errorMessage)
+		}
+		return nil, errors.Errorf("failed to read manifest file from Confluent Hub")
+	}
+
+	pluginManifest := new(ccstructs.Manifest)
+	if err := json.Unmarshal(body, &pluginManifest); err != nil {
+		return nil, err
+	}
+
+	return pluginManifest, nil
+}
+
+func (c *Client) GetRemoteArchive(pluginManifest *ccstructs.Manifest) ([]byte, error) {
+	req, err := http.NewRequest(http.MethodGet, pluginManifest.Archive.Url, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	if c.Debug {
+		dump, err := httputil.DumpRequestOut(req, true)
+		if err != nil {
+			return nil, err
+		}
+		log.CliLogger.Tracef("\n%s\n", string(dump))
+	}
+
+	r, err := c.Client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	if r.StatusCode != http.StatusOK {
+		return nil, errors.New("failed to retrieve archive from Confuent Hub")
```

Comment:
Since we're naming this command `confluent connect plugin install`, I think we should fully move away from using the name "Confluent Hub".

```suggestion
		return nil, errors.New("failed to retrieve plugin archive")
```

Reply:
It's still called Confluent Hub in the UI, so for now I think we should keep references to Confluent Hub. At the very least, there's another reference to Confluent Hub in a suggestion string explicitly telling the user to get the id from Confluent Hub that we should keep if nothing else.